docs/source/examples/flow_past_cylinder.rst

```diff
@@ -0,0 +1,252 @@
+* Creating ghost particles
+* Creating updater
+* Overall setup
+* Evaluating forces on solid
+
+Construction of IOM
+~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code:: python
+
+    def _get_io_info(self):
+        from pysph.sph.bc.hybrid.inlet import Inlet
+        from pysph.sph.bc.hybrid.outlet import Outlet
+        from pysph.sph.bc.hybrid.simple_inlet_outlet import (
+            SimpleInletOutlet)
+        i_update_cls = Inlet
+        o_update_cls = Outlet
+        o_has_ghost = False
+        manager = SimpleInletOutlet
+        props_to_copy += ['uta', 'pta', 'u0', 'v0', 'w0', 'p0']
+
+        inlet_info = InletInfo(
+            pa_name='inlet', normal=[-1.0, 0.0, 0.0],
+            refpoint=[0.0, 0.0, 0.0], equations=inleteqns,
+            has_ghost=i_has_ghost, update_cls=i_update_cls,
+            umax=umax
+            )
+
+        outlet_info = OutletInfo(
+            pa_name='outlet', normal=[1.0, 0.0, 0.0],
+            refpoint=[self.Lt, 0.0, 0.0], has_ghost=o_has_ghost,
+            update_cls=o_update_cls, equations=None,
+            props_to_copy=props_to_copy
+        )
+
+        return inlet_info, outlet_info, manager
+
+    def _create_inlet_outlet_manager(self):
+        inlet_info, outlet_info, manager = self._get_io_info()
+        iom = manager(
+            fluid_arrays=['fluid'], inletinfo=[inlet_info],
+            outletinfo=[outlet_info]
+        )
+        return iom
+
+In the function ``_get_io_info`` the ``inlet_info`` and ``outlet_info`` are
+created, and manager class are returned. The ``inlet_info`` and ``outlet_info``
+info contains specific information about inlet and outlet that enables IOM to
+create equations, stepper and updater. In ``_create_inlet_outlet_manager``
+the IOM is created using the info objects.
+
+Passing IOM to scheme
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code:: python
+
+    def configure_scheme(self):
+        scheme = self.scheme
+        self.iom = self._create_inlet_outlet_manager()
+        scheme.inlet_outlet_manager = self.iom
+        pfreq = 100
+        kernel = QuinticSpline(dim=2)
+        self.iom.update_dx(self.dx)
+        scheme.configure(h=self.h, nu=self.nu)
+
+        scheme.configure_solver(kernel=kernel, tf=self.tf, dt=self.dt,
+                                pfreq=pfreq, n_damp=0)
+
+
+The IOM object of the application is initialized in the method
+``configure_scheme`` of the ``Application`` class. All the
+post-initialization method could be called here e.g. ``update_dx``.
+
+Creating ghost particles
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this example, we have a wind tunnel with two bounding slip walls on the
+top and below of the tunnel. The inlet is on the left and the outlet is on
+the right. In order to perform the simulation five particle arrays,
+``solid``, ``fluid``, ``wall``, ``inlet`` and ``outlet`` are generated.
+``fluid``, ``solid`` and ``wall`` has to obey ``edac`` scheme, whereas
+``inlet`` and ``outlet`` are solved according to the boundary conditions.
```

Comment:
This is a good starting paragraph, maybe it should start with this and then go into IOM details?

Reply:
done